Src/EventCounterCollector/EventCounterCollector/EventCounterCollectionModule.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.EventCounterCollector
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Text;
+    using Microsoft.ApplicationInsights;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.Extensibility.EventCounterCollector.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    /// <summary>
+    /// Telemetry module for collecting EventCounters.
+    /// https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Tracing/documentation/EventCounterTutorial.md.
+    /// </summary>
+    public class EventCounterCollectionModule : ITelemetryModule, IDisposable
+    {
+        private readonly double refreshInternalInSecs = 60;
+
+        /// <summary>
+        /// TelemetryClient used to send data.
+        /// </summary>        
+        private TelemetryClient client = null;        
+        private EventCounterListener eventCounterListener;
+        private bool disposed = false;
+        private bool isInitialized = false;        
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventCounterCollectionModule"/> class.
+        /// </summary>
+        public EventCounterCollectionModule()
+        {
+            this.Counters = new List<EventCounterCollectionRequest>();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventCounterCollectionModule"/> class.
+        /// </summary>
+        internal EventCounterCollectionModule(double refreshIntervalInSecs) : this()
+        {
+            this.refreshInternalInSecs = refreshIntervalInSecs;
+        }
+
+        /// <summary>
+        /// Gets the list of counter names to collect. Each should have the name of EventSource publishing the counter, and counter name.
+        /// </summary>
+        public IList<EventCounterCollectionRequest> Counters { get; private set; }
```

Comment:
any reason for IList? Could it be IEnumerable?

Reply:
Could be. Followed same approach as existing performance counter modul.

Reply:
Actually we use count so IList is needed.